### PR TITLE
wip REF: redirect `df.to_html` to Styler if new kwargs are used.

### PR DESCRIPTION
**Pull request by attack68**
This is an alternatative way of transitioning `DataFrame.to_html` to `Styler.to_html`.

It is a very minimalist introduction:

  - It will **fallback to the original HTMLFormatter implementation** if:
    - any of the deprecated kwargs are specifically input, and will show a **deprecation warning**.
    - **none** of the new Styler implementation kwargs are input.
  - It will **only use the new Styler implementation** if no deprecated kwargs are input and at least one **new** kwarg is used.

here is the rendered doc-strings:
Screenshot 2022-01-02 at 15 46 11
Screenshot 2022-01-02 at 15 46 23
Screenshot 2022-01-02 at 15 46 34
Screenshot 2022-01-02 at 15 46 45
Screenshot 2022-01-02 at 15 46 56
Screenshot 2022-01-02 at 15 47 07
Screenshot 2022-01-02 at 15 47 17
Screenshot 2022-01-02 at 15 47 25
Screenshot 2022-01-02 at 15 47 32

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2865,16 +2865,29 @@ def to_html(
         justify: str | None = None,
         max_rows: int | None = None,
         max_cols: int | None = None,
-        show_dimensions: bool | str = False,
+        show_dimensions: bool | str | None = None,
```
Comment (jreback) [inline pandas/core/frame.py:2868]: dont' you want these as `lib.no_default`? (e.g. to see if the user passed them)?

File: pandas/core/frame.py [hunk 2]
```diff
@@ -2865,16 +2865,29 @@ def to_html(
...
         escape: bool = True,
-        notebook: bool = False,
+        notebook: bool | None = None,
         border: int | None = None,
         table_id: str | None = None,
-        render_links: bool = False,
+        render_links: bool | None = None,
         encoding: str | None = None,
+        *,
+        table_attributes: str | None = None,
```
Comment (jreback) [inline pandas/core/frame.py:2879]: why are you adding the *new* options?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 2 above)
Comment (attack68) [reply pandas/core/frame.py:2879]: this suggestion by @jorisvandenbossche https://github.com/pandas-dev/pandas/pull/44451#issuecomment-1000741346

File: pandas/core/frame.py (hunk 1 above)
Comment (attack68) [reply pandas/core/frame.py:2868]: probably, I've not really used `lib.no_default` before and a previous suggestion was to use `None`. Happy to investigate and switch if preferred.

Comment (jorisvandenbossche) [reply pandas/core/frame.py:2858]: `None` is fine to use for this purpose, as long as it is not a valid value for the keyword

**Comment by datapythonista** [comment]
@attack68 do you still want to work on this?

**Comment by attack68** [comment]
@datapythonista I do but it requires some pre requisites and a community consensus. There is a bit more progress on `DataFrame.to_latex`. Hopefully in future will try to use that as a template, so will close this for now.

**Pull request closed without merging**